validphys2/src/validphys/theorycovariance/construction.py

```diff
@@ -199,6 +225,234 @@ def covmat_9pt(name1, name2, deltas1, deltas2):
+
+    # ABMP parametrisation
+    x = [0.0, 0.1, 0.3, 0.5, 0.7, 0.9, 1]
+    y_2 = [0.023, -0.032, -0.005, 0.025, 0.051, 0.003, 0.0]
+    y_T = [-0.319, -0.134, -0.052, 0.071, 0.030, 0.003, 0.0]
+    H_2 = scint.CubicSpline(x, y_2)
+    H_2 = np.vectorize(H_2)
+    H_T = scint.CubicSpline(x, y_T)
+    def H_L(x):
+        return np.power(x, 0.05)*(H_2(x) - H_T(x))
+    H_L = np.vectorize(H_L)
+
+    for proc in process_info.namelist.keys():
+        running_index_proc = 0
+        x  = np.array([])
+        Q2 = np.array([])
+        y  = np.array([])
+        N = np.array([])
+
+        for exp in process_info.namelist[proc]:
+            # Locate position of the experiment
+            size = process_info.sizes[exp]
+            start_proc_by_exp[exp] = running_index_tot
+            running_index_tot += size
+            running_index_proc += size
+
+            # Compute shifts only for a subset of processes
+            if proc in filtered_proc and exp not in excluded_exp:
+                #central = process_info.preds[proc][1][start_proc_by_exp[exp] : size] # Probably this is deprecated
+                x = process_info.data[proc].T[0][running_index_proc - size : running_index_proc]
+                Q2 = process_info.data[proc].T[1][running_index_proc - size : running_index_proc] 
+                y = process_info.data[proc].T[2][running_index_proc - size : running_index_proc]
+
+                # TO UNDERSTAND: how can I retrieve the observable from the name of the experiment
+                if "SIGMA" in exp:
+                    N_2, N_L = compute_normalisation_by_experiment(exp, x=x, y=y, Q2=Q2)
+
+                elif "F2":
+                    N_2 = np.ones(shape=x.shape)
+                    N_L = np.zeros(shape=x.shape)
+
+                else:
+                    raise ValueError(f"The normalisation for the observable ... is not known.")
+
+                if ht_pt_prescription == 5:
+                    deltas["(+,0)"] += [N_2 * ht_coeff_1 * H_2(x)]
+                    deltas["(0,+)"] += [N_L* ht_coeff_2 * H_L(x)]
+                else:
+                    raise ValueError(
+                        f"The pt prescription for the HT theory covmat is not supported."
+                    )
+
+    # Construct theory covmat
+    covmats = defaultdict(list)
+    for proc1 in filtered_proc:
+        for proc2 in filtered_proc:
+            for i, exp1 in enumerate(process_info.namelist[proc1]):
+                for j, exp2 in enumerate(process_info.namelist[proc2]):
+                    if ht_pt_prescription == 5:
+                        s = np.outer(deltas["(+,0)"][i], deltas["(+,0)"][j]) + \
+                            np.outer(deltas["(0,+)"][i], deltas["(0,+)"][j])
+                    start_locs = (start_proc_by_exp[exp1], start_proc_by_exp[exp2])
+                    covmats[start_locs] = s
+    return covmats
+
+
+def compute_normalisation_by_experiment(experiment_name, **kwargs):
```

Comment:
Why are you collecting them as kwargs if they are being used in the funtion?

P.S. python is not cpp, to keep things more readable it helps to be consistent and e.g. use `**` for power and `dictname['keyname']` everywhere